app/main.go

```diff
@@ -0,0 +1,24 @@
+package app
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/pkg/errors"
+)
+
+func Main(main func(context.Context) error) {
+	if err := run(main); err != nil && errors.Cause(err) != context.Canceled {
```

Comment:
What about `context.DeadlineExceeded`? It's probably something we should echo out; just wondering if that was a choice or overlooked.

Reply:
It's on purpose. Program should not exit because of a timeout on a context. Not a controller for sure.